components/org.wso2.carbon.identity.remotefetch.core/src/main/java/org/wso2/carbon/identity/remotefetch/core/executers/tasks/RemoteFetchConfigurationImmediateTask.java

```diff
@@ -0,0 +1,155 @@
+import org.wso2.carbon.identity.remotefetch.common.RemoteFetchConfiguration;
+import org.wso2.carbon.identity.remotefetch.common.actionlistener.ActionListener;
+import org.wso2.carbon.identity.remotefetch.common.actionlistener.ActionListenerBuilder;
+import org.wso2.carbon.identity.remotefetch.common.actionlistener.ActionListenerBuilderException;
+import org.wso2.carbon.identity.remotefetch.common.actionlistener.ActionListenerComponent;
+import org.wso2.carbon.identity.remotefetch.common.configdeployer.ConfigDeployer;
+import org.wso2.carbon.identity.remotefetch.common.configdeployer.ConfigDeployerBuilder;
+import org.wso2.carbon.identity.remotefetch.common.configdeployer.ConfigDeployerBuilderException;
+import org.wso2.carbon.identity.remotefetch.common.configdeployer.ConfigDeployerComponent;
+import org.wso2.carbon.identity.remotefetch.common.exceptions.RemoteFetchCoreException;
+import org.wso2.carbon.identity.remotefetch.common.repomanager.RepositoryManager;
+import org.wso2.carbon.identity.remotefetch.common.repomanager.RepositoryManagerBuilder;
+import org.wso2.carbon.identity.remotefetch.common.repomanager.RepositoryManagerBuilderException;
+import org.wso2.carbon.identity.remotefetch.common.repomanager.RepositoryManagerComponent;
+import org.wso2.carbon.identity.remotefetch.core.internal.RemoteFetchServiceComponentHolder;
+
+/**
+ * This class provide immediate task to be executed for trigger call of given remote fetch configuration.
+ * It retrieves relevant action lister component, deployer component and repository manager component using given
+ * remote fetch configuration object from componant registry.
+ * It builds action listener, deployer and repository manager using relevant builders
+ * It runs corresponding action listener iteration. *
+ */
+public class RemoteFetchConfigurationImmediateTask implements Runnable {
+
+    private static final Log log = LogFactory.getLog(RemoteFetchConfigurationImmediateTask.class);
+
+    private RemoteFetchComponentRegistry componentRegistry = RemoteFetchServiceComponentHolder.getInstance()
+            .getRemoteFetchComponentRegistry();
+
+    private RemoteFetchConfiguration remoteFetchConfiguration;
+
+    public RemoteFetchConfigurationImmediateTask(RemoteFetchConfiguration remoteFetchConfiguration) {
+
+        this.remoteFetchConfiguration = remoteFetchConfiguration;
+    }
+
+    /**
+     * This method is used to retrieve relevant components and build lister using given remote fetch configuration.
+     * @return ActionListener
+     * @throws RemoteFetchCoreException
+     */
+    private ActionListener buildListener() throws RemoteFetchCoreException {
+
+        RepositoryManager repositoryManager;
+        ActionListener actionListener;
+        ConfigDeployer configDeployer;
+
+        // Get an instance of RepositoryManager from registry.
+        RepositoryManagerComponent repositoryManagerComponent = this.componentRegistry
```

Comment:
We need to refactor this long method to few logical private methods for better readability and maintainability.